librecad/src/lib/information/rs_information.cpp

```diff
@@ -143,6 +144,31 @@ bool RS_Information::isTrimmable(RS_Entity* e1, RS_Entity* e2) {
     return false;
 }
 
+RS_Entity* RS_Information::getRadiusArc(RS_Entity* e1, RS_Entity* e2) {
+	if (!e1 || !e2)
+		return nullptr;
+	if (!e1->getParent() || !e2->getParent())
+		return nullptr;
+	if (e1->getParent()->rtti()==RS2::EntityPolyline &&
+		e2->getParent()->rtti()==RS2::EntityPolyline &&
+		e1->getParent()==e2->getParent()) {
+		RS_Polyline* poly = (RS_Polyline*)e1->getParent();
+		int idx1 = poly->findEntity(e1);
+		int idx2 = poly->findEntity(e2);
+		unsigned dist = abs(idx1 - idx2);
+		int mid = -1;
+		if (dist == 2)
+			mid = idx1 + (idx2 - idx1) / 2;
+		else if (poly->isClosed() && dist == poly->count() - 2U)
+			mid = (std::max(idx1, idx2) + 1) % poly->count();
```

Comment:
a little bit confusion here for closed polylines:

```
0 - 1
|   |
3 - 2
```

given idx1=3, idx2=1, this algorithm gives mid = 0, seems to be perfect
but how could I handle idx1=1, idx2=3, and expected mid=1

```C++
if (idx2 < idx1) idx2 += poly->count();
mid = (idx1 + (idx2 - idx1)/2)%poly->count();
```